rq/worker.py

```diff
@@ -1728,6 +1728,7 @@ def fork_work_horse(self, job: 'Job', queue: 'Queue'):
         """Spawns a work horse to perform the actual work using os.spawn()."""
         os.environ['RQ_WORKER_ID'] = self.name
         os.environ['RQ_JOB_ID'] = job.id
+        connection_kwargs = {k: v for k, v in self.connection.connection_pool.connection_kwargs.items() if k != 'retry'}
```

Comment:
Do you mind removing this workaround and rebase this PR to follow master? I decided to disallow redis-py 6.0 in this PR: https://github.com/rq/rq/pull/2244

RQ should work with redis-py >= 6.0.1 since what caused this error has been fixed in redis-py's master branch.